### PR TITLE
reading hostname from resourceconfig file

### PR DESCRIPTION
**Pull request by mvsusp**
*Description of changes:*
- reading hostname from resourceconfig file

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit/c/test_libchangehostname.py [hunk 1]
```diff
@@ -10,22 +10,44 @@
 # distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
+# import json
```
Comment (uditbhatia) [inline test/unit/c/test_libchangehostname.py]: Remove unwanted imports

File: src/sagemaker_containers/c/libchangehostname.c [hunk 2]
```diff
@@ -12,23 +12,81 @@
...
+	/* Assume the top-level element is an object */
+	if (r < 1 || token[0].type != JSMN_OBJECT) {
+		printf("Object expected\n");
+		return 1;
+	}
 
-static PyObject* libchangehostname_call(PyObject* self, PyObject* args) {
+    /* Loop over all keys of the root object */
+    int i = 1;
+    for (; i < r; i++)
```
Comment (laurenyu) [inline src/sagemaker_containers/c/libchangehostname.c]: why not combine ll. 65-66?

Comment (mvsusp) [reply src/sagemaker_containers/c/libchangehostname.c]: Unfortunately, it is against C99 conventions to initialize a variable inside a for loop.

File: src/sagemaker_containers/c/libchangehostname.c [hunk 3]
```diff
@@ -12,23 +12,81 @@
...
-  return 0;
-}
+    fseek(file, 0, SEEK_END);
+
+    long length = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    char * json_string = malloc(length);
+    fread(json_string, 1, length, file);
+    json_string[length] = '\0';
```
[the remark this answers is not in the document]
Comment (mvsusp) [reply src/sagemaker_containers/c/libchangehostname.c]: I'm appending the NULL character to represent the end of the string.

File: src/sagemaker_containers/c/libchangehostname.c [hunk 4]
```diff
@@ -12,23 +12,81 @@
...
 
 #include <Python.h>
+#include <stdio.h>
 #include <stdlib.h>
+#include "jsmn.h"
 
+static int jsoneq(const char *json, jsmntok_t *tok, const char *s)
+{
+    if (tok->type == JSMN_STRING && (int)strlen(s) == tok->end - tok->start &&
+        strncmp(json + tok->start, s, tok->end - tok->start) == 0)
```
[the remark this answers is not in the document]
Comment (mvsusp) [reply src/sagemaker_containers/c/libchangehostname.c]: Good call, I wrote a test showing that it is not an issue.

File: src/sagemaker_containers/c/libchangehostname.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (laurenyu) [reply src/sagemaker_containers/c/libchangehostname.c]: shouldn't that be at index `length+1` then? (or is there a EOF character you're overriding?)

File: src/sagemaker_containers/c/libchangehostname.c (hunk 2 above)
Comment (laurenyu) [reply src/sagemaker_containers/c/libchangehostname.c]: ah, that's right. I think I'm used to seeing it as:

```c
int i;
for (i = 1; i < r; i++)
```

but I don't have a strong opinion about it

File: src/sagemaker_containers/c/libchangehostname.c [hunk 5]
```diff
@@ -12,23 +12,83 @@
...
 
-  strncpy(name, val, len);
-  return 0;
-}
+    fseek(file, 0, SEEK_END);
+
+    long length = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    char *json_string = malloc(length +1);
```
Comment (laurenyu) [inline src/sagemaker_containers/c/libchangehostname.c]: nit: space after the "+"

File: src/sagemaker_containers/c/libchangehostname.c [hunk 6]
```diff
@@ -12,23 +12,83 @@
...
+            strncpy(name, val, len);
+
+            name[len] = "\0";
+
+            free (json_string);
+            return 0;
+        }
+    }
+
+    free (json_string);
```
Comment (laurenyu) [inline src/sagemaker_containers/c/libchangehostname.c]: nit: no space after "free"

**Pull request merged**